### PR TITLE
Hotfix/fix primary available dependency

### PR DESCRIPTION
**Pull request by arbulu89**
Cover the scenario where the secondary node is created before the primary and add a timeout to configure this timeout

### CODE REVIEW

File: hana/primary_available.sls [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{%- from "hana/map.jinja" import hana with context -%}
+{% set host = grains['host'] %}
+
+{% for node in hana.nodes %}
+{% if node.host == host and node.secondary is defined %}
+netcat-openbsd:
```
Comment (krig) [inline hana/primary_available.sls]: actually I think I have a better way to do this in hacluster-formula where it looks for Hawk to be running, no need for netcat: https://github.com/SUSE/habootstrap-formula/blob/master/cluster/join.sls#L3

Comment (arbulu89) [reply hana/primary_available.sls]: Agree! The other way looks better, less dependencies. I will make the update and test it.
Thanks @krig

Comment (krig) [reply hana/primary_available.sls]: ah but couldn't you use the 7630 port here too?

Comment (krig) [reply hana/primary_available.sls]: aha no you need to check if HANA is running of course, sorry!

Comment (arbulu89) [reply hana/primary_available.sls]: Exactly, we need to check is System replication of HANA is enabled. There are many ports for that, but none of them uses http protocol

Comment (krig) [reply hana/primary_available.sls]: I see yeah. Hm, seems salt doesn't come with anything better, at least at a glance..

File: hana/primary_available.sls [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  pkg.installed
+
+primary-available:
+  cmd.run:
+    - name: until nc -z {{  node.secondary.remote_host }} 40002; do sleep 1; done
```
Comment (diegoakechi) [inline hana/primary_available.sls]: @arbulu89 The logic here is a little bit more complex. See: https://help.sap.com/viewer/ports/616a3c0b1cc748238de9c0341b15c63c.html

The port calculation depends if HANA is running on a Single Container or Multi-Container installation, also depends on the instance number like `4<InsNr>02`.

Maybe @Simranpal knows a command or a metadata file to check the port in usage without needing to calculate that.

Comment (arbulu89) [reply hana/primary_available.sls]: True, I didn't notice this logic. Let's see if we can find a better solution to find the port.
Thx @diegoakechi

Comment (diegoakechi) [reply hana/primary_available.sls]: As discussed, we will solve this with a different approach. Approving for now.

**Pull request merged**